src/common/file_system.cpp

```diff
@@ -295,8 +295,120 @@ string FileSystem::GetWorkingDirectory() {
 #endif
 
 string FileSystem::JoinPath(const string &a, const string &b) {
-	// FIXME: sanitize paths
-	return a.empty() ? b : a + PathSeparator(a) + b;
+	auto has_uri_scheme = [](const string &path) {
+		if (path.empty()) {
+			return false;
+		}
+		if (StringUtil::StartsWith(path, "file:")) {
+			return true;
+		}
+		auto scheme_pos = path.find("://");
+		if (scheme_pos == string::npos) {
+			return false;
+		}
+		// avoid treating a Windows drive ("C:\") as a scheme
+		if (scheme_pos == 1 && StringUtil::CharacterIsAlpha(path[0])) {
```

Comment:
Is this necessary? We are looking for `scheme://` above, are we treating `C://` as a Windows drive? Shouldn't that be only `C:\`?